File: WEEK_5_C/Q2.c
#include <stdio.h>
#include <stdlib.h>

struct Node{ int data; struct Node* next; };

int main(){
    struct Node *top=NULL,*t;
    t=(struct Node*)malloc(sizeof(struct Node)); t->data=5; t->next=top; top=t;
    t=(struct Node*)malloc(sizeof(struct Node)); t->data=10; t->next=top; top=t;
    printf("Pop=%d\n",top->data); t=top; top=top->next; free(t);
    printf("Top=%d\n",top->data);
    return 0;
}

File: WEEK_5_Python/Q2.py
class Node:
    def __init__(self, data): self.data, self.next = data, None

class Stack:
    def __init__(self): self.head = None
    def push(self, x):
        node = Node(x); node.next = self.head; self.head = node
    def pop(self):
        if not self.head: return None
        val = self.head.data; self.head = self.head.next; return val
    def top(self): return self.head.data if self.head else None

# Example
s = Stack()
s.push(5); s.push(7)
print(s.pop())  
print(s.top())  
